packages/swap/src/utils/promise.ts

```diff
@@ -9,3 +9,22 @@ export const deferredPromise = <T>() => {
 
   return { promise, resolve: resolve!, reject: reject! };
 };
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type AsyncFn = (...args: any[]) => Promise<any>;
+
+type AsyncReturnType<F extends AsyncFn> = Awaited<ReturnType<F>>;
+
+/**
+ * use this to wrap an async function to return a result object so no uncaught
+ * promise rejections are thrown if you do not immediately await/handle it
+ * @param fn an asynchronous function
+ */
+export const resultify =
```

Comment:
Well it wraps a function rather than a promise, so the naming convention is a mixture of something like `promisify` from `util` and `Result` which is what (at least the ts team calls the return values from `Promise.allSettled`
<img width="342" alt="image" src="https://github.com/chainflip-io/chainflip-sdk-monorepo/assets/21286068/13c61337-0f66-46f7-aa4c-cab036df418c">